src/utils/url.js

```diff
@@ -96,7 +96,6 @@ export const tryParseUrl = (url: string, base?: string | URL): URL | void => {
  *
  * (If relative, we assume callers want to treat it relative to the realm.)
  */
-// TODO: Careful! Gives a false positive and false negatives; see comments.
```

Comment:
Ah, self-nit: one identified false negative remains in this commit, so don't remove this whole TODO.